frontend/src/components/courses/evaluations/Evaluate.tsx

```diff
@@ -157,30 +155,24 @@ export const QuestionsForm = ({ questions, course }) => {
             as={Card}
           >
             {questions.map((question) => {
-              if (question.archived) return null;
-
-              let field: JSX.Element = <p>Error</p>;
-              if (question.category === "R") {
-                field = (
-                  <RatingField
-                    question={question}
-                    id={question.id}
-                    name="ratings"
-                    label="First Name"
-                    {...props}
-                  />
-                );
-              } else if (question.category === "C") {
-                field = (
-                  <CommentField
-                    question={question}
-                    id={question.id}
-                    name="comments"
-                    label="First Name"
-                    {...props}
-                  />
-                );
-              }
+              const field = question.archived ? null : question.category ===
```

Comment:
Field becomes an  unnecessary intermediary here, you can directly inject this where you call `{field}`